nixos/modules/services/networking/dnschain.nix

```diff
@@ -136,10 +136,18 @@ in
         "/.dns/127.0.0.1#${toString cfg.dns.port}"
       ];
 
-    services.pdns-recursor.forwardZones = mkIf cfgs.pdns-recursor.resolveDNSChainQueries
-      { bit = "127.0.0.1:${toString cfg.dns.port}";
-        dns = "127.0.0.1:${toString cfg.dns.port}";
-      };
+    services.pdns-recursor = mkIf cfgs.pdns-recursor.resolveDNSChainQueries {
+      forwardZones =
+        { bit = "127.0.0.1:${toString cfg.dns.port}";
+          dns = "127.0.0.1:${toString cfg.dns.port}";
+        };
+      extraConfig =''
+        lua-config-file=${pkgs.writeText "lua.conf" ''
```

Comment:
Do multiple assignments of lua-config-file get merged together? Because if not this is a minor backwards incompatibility for when it gets assigned somewhere else as well

Reply:
Good question. I don't think so: probably the last declaration takes precedence.

Reply:
That's actually a good demonstration for https://github.com/NixOS/rfcs/pull/42 as to why `extraConfig`s are bad, because in this case this causes a backwards incompatibility really hard to work around.

Do you think it's worth introducing an additional option like `luaConfigText` to let users at least work around this? `lua-config-file` assignments by users would have to be manually changed to be in `luaConfigText`, which gets merged by concatenation into a single file for the final `lua-config-file` setting.

Reply:
It would make sense since the scope of the lua configuration is much larger, though it's probably uncommon.

Reply:
Anyway, I like your proposal: do you think I should try implementing a .settings for pdns-recursor?
The format is pretty simple so a json/attrs set should be enough.

Reply:
That would be great yeah, you could even deprecate the `extraConfig` option in favor of `settings`. With only such a new option, it would be possible to (at runtime) generate a file by concatenating all `lua-config-file` assignments if you feel like implementing it.

Otherwise a `luaConfigText` option would work too, which just sets `lua-config-file = mkDefault (pkgs.writeText "config.lua" luaConfigText)`, meaning assignments to `lua-config-file` by the user will be able to override this.

Reply:
I'm not sure if you got the notification because of another github outage: I have made a PR implementing the `settings` options: #67251.